storm-server/src/main/java/org/apache/storm/metric/StormMetricsRegistry.java

```diff
@@ -53,12 +55,14 @@ public static Meter registerMeter(String name) {
      *
      * @param topoConf config that specifies reporter plugin
      */
-    public static void startMetricsReporters(Map<String, Object> topoConf) {
-        for (PreparableReporter reporter : MetricsUtils.getPreparableReporters(topoConf)) {
+    public static AutoCloseable startMetricsReporters(Map<String, Object> topoConf) {
```

Comment:
I think it would be better to use another interface that extends AutoCloseable so you don't have to deal with the non-existing Exception everywhere.

Reply:
I guess we can use Runnable instead. But it doesn't look as semantically correct as Autocloseable here (as we're closing the reporters)

Reply:
No, I meant declare a new interface that extends AutoCloseable but doesn't throw Exception

```
interface NotThrowingAutoCloseable extends AutoCloseable {
  void close();
}
```

Reply:
Okay. Do we want this to be a generic utility interface or specific to reporters then?

Reply:
I'd probably just declare it in this file, we can always move it later if we need to. If we make the registry non-static at some point, we probably won't need it anymore, since we can just add a close method to the registry instead.